dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/servlet/filter/OpenAPIEndpointFilter.java

```diff
@@ -52,15 +53,16 @@ public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain
             chain.doFilter(req, wrapper);
 
             //Modify Response
-            if (resp.getContentType() != null && resp.getContentType().contains("text/html")) {
+            if (httpServletResp.getContentType() != null && httpServletResp.getContentType().contains("text/html")) {
                 String content = wrapper.getContentAsString();
                 // in case we have had a issue creating the tracker earlier - if so leave content as-is
                 if(openAPIEndpointTracker != null) {
                     // replace the default URL for the document endpoint with the value that is being used
                     content = content.replaceAll("/openapi", openAPIEndpointTracker.getService().getOpenAPIDocUrl());
                 }
-                resp.setContentLength(content.length());
-                resp.getWriter().write(content);
+                httpServletResp.setHeader("Content-Type",httpServletResp.getContentType()+";charset=UTF-8");
+                httpServletResp.setContentLength(content.getBytes(StandardCharsets.UTF_8).length);
+                httpServletResp.getWriter().write(content);
             }
         }
```

Comment:
I have a second concern about this as well.

The HTML is being read from a file, presumably as bytes. In this filter, we're then asking for the content as a string.

Given the problems we have had have been down to the servlet container defaulting to ISO-8859-1, is the servlet container interpreting the bytes from the file as ISO-8859-1 rather than UTF-8?

This may not cause problems with our specific files if they're entirely ascii, but it does suggest we're doing something wrong here that may fail in the future.

Reply:
Possibly the right thing to do is to get and set bytes and do any encoding ourselves, relying on the fact that we know the source files are UTF-8.

Possibly we also want a better check that we're only operating on the index.html file.

Reply:
Have changed getContentAsString with getContentAsBytes and now construct the String with that using UTF-8 to reduce any potential ISO issues.